internal/core/docker_host.go

```diff
@@ -56,6 +56,23 @@ func DefaultGatewayIP() (string, error) {
 	return ip, nil
 }
 
+// defaultCallbackCheckFn Use a vanilla Docker client to check if the Docker host is reachable.
+// It will avoid recursive calls to this function.
+var defaultCallbackCheckFn = func(ctx context.Context, host string) error {
```

Comment:
nit: should we change the name of this so it better describes its new behaviour, its not a default any more.
```suggestion
// dockerHostCheck validates that host is reachable.
var dockerHostCheck = func(ctx context.Context, host string) error {
```

Reply:
Done